pyomo/repn/plugins/nl_writer.py

```diff
@@ -1957,7 +2021,16 @@ def _before_npv(visitor, child):
     # else:
     #     child = visitor.value_cache[_id] = child()
     # return False, (_CONSTANT, child)
-    return False, (_CONSTANT, child())
+    try:
+        tmp = False, (_CONSTANT, child())
+        if tmp[1][1].__class__ is complex:
+            return True, None
```

Comment:
It's mostly so that I don't have to repeat the error handling/message about Pyomo not handling complex numbers (and the other location gives better context, because it is at the node that generated the complex number)